### PR TITLE
Added build dependencies

### PR DESCRIPTION
**Pull request by carldebilly**

### CODE REVIEW

File: src/Uno.SourceGeneration/SourceGeneratorDependencyAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	[AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
+	public class SourceGeneratorDependencyAttribute : Attribute
+	{
+		/// <summary>
+		/// Fully Qualified Name (FQN: namespace + class name) of the generator to execute before.
+		/// </summary>
+		/// <remarks>
+		/// No effect if the generator is not found.
+		/// </remarks>
+		public string GeneratorToExecuteBefore { get; }
```
Comment (jeromelaban) [inline src/Uno.SourceGeneration/SourceGeneratorDependencyAttribute.cs]: DependsOn?

File: src/Uno.SourceGenerationHost.Shared/SourceGeneratorHost.cs [hunk 2]
```diff
@@ -64,92 +65,140 @@ public string[] Generate()
...
 
-			var generatorResults = details.Generators
-				.AsParallel()
-				.Select(generatorDef =>
+			// Build dependencies graph
+			var generatorsByName = details.Generators.ToDictionary(g => g.generatorType.FullName);
+			var generatorNames = generatorsByName.Keys;
+
+			// Dependencies list, in the form (before, after)
+			var dependencies = details.Generators
```
Comment (jeromelaban) [inline src/Uno.SourceGenerationHost.Shared/SourceGeneratorHost.cs]: Indent

File: src/Uno.SourceGeneration/SourceGeneratorDependencyAttribute.cs (hunk 1 above)
Comment (carldebilly) [reply src/Uno.SourceGeneration/SourceGeneratorDependencyAttribute.cs]: Yeah your name is probably better.

File: src/Uno.SourceGenerationHost.Shared/SourceGeneratorHost.cs (hunk 2 above)
Comment (carldebilly) [reply src/Uno.SourceGenerationHost.Shared/SourceGeneratorHost.cs]: I don't understand.

**Pull request merged**